Blish HUD/GameServices/Input/Mouse/MouseHandler.cs

```diff
@@ -25,6 +25,8 @@ public class MouseHandler : IInputHandler {
         /// </summary>
         public Point Position => this.State.Position;
 
+        public Point PositionRaw { get; private set; }
```

Comment:
Seems redundant to have an accurate position and a non-accurate position.  Should we just use this as the base of our positions?

`Position` needs to be scaled value as that is used for UI positioning.

If not, at the least I'd like to have `PositionRaw` flipped around to `RawPosition` or better yet `ScreenPosition` or `DesktopPosition`.

Reply:
I had a look at our current method of handling input and saw 3 options:

1. dump `GetState()` and use the windows input event handling system (my original plan)
To keep it non-breaking, I would have had to make a bigger change to the whole handling, since in its current form, this would result in a lot of funky behavior. In the update loop, we only process the last event sent to us, which is fine, since we filter for mouse clicks currently and they are spaced out wide enough for us to catch every single one of them. Mouse move events would shadow clicks easily this way.
After evaluating the impact of `GetState()`, I figured if it would be faster at all, the change wouldn't be noticeable.
2. dump `GetState()`, but just ignore everything except left mouse click, right mouse click and scroll wheel. This is the breaking change I mentioned earlier.
3. Let everything be what it currently is and just expose the raw values for anyone who needs them.

I am interest in your opinion here.

Reply:
I dislike item 3 simply because it adds an extra value to decide if it should be used and trying to explain when it would be best to use one or the other seems like a hard line to draw.  I imagine most folks would want the more raw and accurate value.

For instance, is there something specifically wrong with using:

```cs
this.State = new MouseState((int)(this.PositionRaw.X / GameService.Graphics.UIScaleMultiplier),
							(int)(this.PositionRaw.Y / GameService.Graphics.UIScaleMultiplier),
							_mouseEvent?.WheelDelta ?? 0,
							rawMouseState.LeftButton,
							rawMouseState.MiddleButton,
							rawMouseState.RightButton,
							rawMouseState.XButton1,
							rawMouseState.XButton2);
```

Perhaps I am missing where the breaking change is that you're mentioning.

Reply:
Where does rawMouseState come from? The event does not provide the full mouse state, but only the fraction that changed. So a mouse move event only contains the new position of the mouse.

Reply:
It's defined a few lines above the `this.State` assignment.

Reply:
I am a little confused. What I see is this:
```cs
            MouseState prevMouseState = this.State;

            MouseState rawMouseState = Mouse.GetState();
```
which gets `rawMouseState` from the `GetState()` method. Did I understand you correctly that you want to get rid of `Mouse.GetState()`?

Reply:
Not necessarily the entire `GetState()` since we're using it for those other button states.  Ideally we would have only one source and reference for the mouse position, so if we have a more accurate or "raw" mouse position, I would think it would be more ideal to use that instead of also using the polled value from `GetState()`.  That way we don't need `.PositionRaw` and can instead use `.Position` (which is just `this.State.Position`) for everything.

Reply:
Oh okay, that's no problem.